### ISSUE TITLE
[Suggestion] Discussion forum other than Github

### ISSUE THREAD

**Issue by ancientcampus** [HabitRPG/habitica#332]
Eventually, I think habitRPG will need its own forum outside of Github. Github is great for bugs and features, but a forum is friendlier to users who want to swap ideas, share success stories, ask newbie questions, coordinate parties, and motivate each other.

I understand that this is low priority, and eventually you'd want the forum to run off your server as opposed to a place like proboards, and so you might not wish to start a free off-site forum and later make everyone switch. That said, I think a free off-site forum would be helpful currently, as I believe an on-site forum will remain low-priority for a while still.

This is great, by the way. The project has me very excited!

**Comment by lefnire** [comment]
there's the [Reddit group](http://www.reddit.com/r/habitrpg), not sure what else people prefer

**Comment by Pandoro** [comment]
Well a real forum would be a nice thing indeed. Then you get user profiles
and post counts etc. Pretty much like github, but without feeling bad about
asking simple questions etc. I once ran a free forum for my guild in some
game. There is a shit load of free forum code out there and a server is
there right? It shouldn't be all that hard to realize.

I'd prefer this over Reddit or any other site already there, as this would
be only for habitrpg and not for ten other things at the same time.
Drawback is of course that people will need to register :x

**Comment by ancientcampus** [comment]
I didn't hear about the Reddit group, nice. There's lots of free code for those with a server, and sites like proboards (a google search will show lots of others) will do all the hosting and everything for free (and keep the ad revenue - though I don't remember any popops or obtrusive adds).

Whatever it is, I think the main advantage would to be able to link it from the main page.

**Comment by lefnire** [comment]
I'm a Drupal developer from a previous life, I could put something together quickish. Will have to wait till after the KS, but noted.

**Comment by StanLindsey** [comment]
Forums are mega easy to set-up, unless you specifically want to develop something due to some of the forum software currently available's complexity.

**Comment by lefnire** [comment]
only thing that might make this complex is linking user accounts to the forum, as in the WoW forums. I quite like the idea, but we might have to skip that

**Comment by Pandoro** [comment]
Honestly, just take some free forum and smack it onto the server :) We can
start writing tutorials and updates there more frequently and compose FAQs
etc. Of course we can do that with a wiki, but that lacks the user
interaction a little. And github is probably a little to techie for most
people and rather restricted to issues and suggestions.  I wouldn't go with
anything to hard core, although the linking would be really nice ^^; If
that has to wait for a longer time though, I'm not sure if it is worth it.
Maybe we could come up with some free forums and you could do a quick
screening if you think it could be integrated later on or not? Or if you
already know something which could support this at some point?

**Comment by ghost** [comment]
I owned a forum a few years ago, this is what we used http://www.mybb.com/.

**Comment by StanLindsey** [comment]
Many sites link between "app" site and "forum" site at later dates. I'm fairly sure it would be easy to add a "user ID&APIToken" in the profile settings which could display their character in their forum profile or something. I mean, what functionality do we need. 

Overall; forums a good idea me tinks. That will seperate github "issues" from the community. Not that we don't love you! 

It does add yet another site for us to check on every hour =P (We'll have to get proper moderators!)

**Comment by Pandoro** [comment]
Well once the KS ends, we don't need to be checking so many sites anymore I
guess? Of course github will stay an issue tracker, but it will maybe be
more of an issue tracker than an "everything" tracker as it is right now.

The forum could be the main thing, for announcement and information about
everything. Would leave the tumblr, which might also be done within the
forum? Within a news section? And of course twitter etc. But overall it
sounds pretty much limited to github and a forum then. Am I missing
something?

**Comment by ghost** [comment]
Maybe we can try this http://www.discourse.org/

**Comment by StanLindsey** [comment]
@Pandoro Yeah I think you're correct actually.

@Neohuman - That is actually genuinely sexy and also the default look suits much of HRPGs themeing. 

@lefnire - Have a look at http://try.discourse.org/

Edit: Except A. its not available officially yet and B. Costs like $20 a month were it to be. 

http://vanillaforums.org/ looks good - free, open source(which we like) & penny arcade use it so it must be good.

**Comment by ghost** [comment]
That is exactly what I was thinking

**Comment by ancientcampus** [comment]
[quote]It does add yet another site for us to check on every hour =P (We'll have to get proper moderators!)[/quote]

Mod: "You have been spending way too much time here, get back to work!" _user has been banned for 24 hours_

**Comment by wc8** [comment]
@ancientcampus: yep; but some of the mods will need to ban themselves sometimes ;-)

**Comment by elssar** [comment]
@SlappyBag Discourse is open source(GPL). They have a [repo](https://github.com/discourse/discourse) on Github.

I think it'll only cost you if you use their servers, but not if host it on your own(other than your hosting costs)

**Comment by ghost** [comment]
If that be the case than the logical choice is discourse.

**Comment by StanLindsey** [comment]
@elssar - great spot there, thanks. I totally missed that by looking at their pricing plans.

**Comment by horusofoz** [comment]
Discourse looks nice.

I suggest Drupal. It's open source, mature code and very customizable.

**Comment by lefnire** [comment]
And I worked as a drupal professional for a long time, easy in. Con is its
another thing to maintain (and the major version upgrades.. Such
difficulty), pro is super customizable and we can use migrate.module for
account replication bw forum and main site

**Comment by tmas** [comment]
I imagine you'll need some kind of web admin on board at some point, so once that person comes on board using Drupal would make a bit more sense. Setting up some kind of forum, even if it was just stock phpBB, would be nice for user discussion of stuff other than bugs and could offload some of the commentary about feature requests from github. If there were community managers, they could compile the ideas from the forums and summarize them on github. Overall, I think a separate section of the site (community.habitrpg.com?) for general user communication and better expression of ideas would be useful. 

EDIT: I don't know about Drupal (I've messed around with it a bit but never got too far), but if there ends up being a phpBB forum I could probably write some kind of profile linking plugin. Should be fairly simple once the API is complete.

**Comment by wizonesolutions** [comment]
For forums, there's now Discourse: http://www.discourse.org/

**Comment by lefnire** [comment]
K, sounds like our most common suggestions are discourse & drupal. Drupal
gives me the most control, which is essential for replicating user accounts
(which won't be available in the first iteration, we just need to get a
forum up stat). But Discourse is hosted (right?) and while not giving us
much control, will at least not burden us with another thing to maintain.
Am I understanding Discourse correctly?

How vital is it at present we replicate user accounts, a la WoW forums?

**Comment by ghost** [comment]
The problem with discourse is that it is still in very active development.
So we would need to be prepared to update the software on a fairly regular
basis.

**Comment by PhilipHow** [comment]
RE: WoW Accounts.. Not vital, but a nice to have, and definitely on an optional opt-in basis.

**Comment by tmas** [comment]
The interesting thing about Discourse is that they're looking for a core
group of 4 forums to help out with testing. It's possible that they'd be
willing to make minor modifications to the software (either to link into
habit directly or to allow someone to write a plugin) if you asked them
about it.

**Comment by ghost** [comment]
Asked them now to wait

**Comment by PhilipHow** [comment]
+1 for Discourse, it looks promising.

Considering the unorthodox platform choices, I say go whole hog and try some funky discussion platforms!

**Comment by Pandoro** [comment]
We discussed this again lately. Based on Tylers strong background we felt that having Drupal is a good thing, as we easily change the theme to match HabitRPG as good as possible and later on integrate the characters easily once time is available. Discourse seems to have a very "own" theme and feeling too it, which we figured might not be the best thing to have while building a reliable community forum. 

Furthermore Discourse has yet to prove itself. Moving a community, if they don't work out, or we decide we need something different seriously will be a pain in the ass! 

Any REAL pros about having discourse over drupal? Apart from being new and fancy and cool ;p ?

**Comment by wizonesolutions** [comment]
Nope, Drupal works too :)

**Comment by elssar** [comment]
@Pandoro I can think of a few reasons -
1. Discourse was built by a bunch of awesome coders specifically for to solve this particular problem.
2. It is a joy to use - after [using](http://try.discourse.org) it, I don't want to go back to using traditional forums.
3. I believe that badges and points will be introduced in the future - so a gamified forum for a gamified habit tracker.
4. Now that I think about it, it is a little like the github issue tracker.(don't know why this is a plus point, except that it is)
5. It is new, and fancy, and cool, and bleeding edge ;p

Having said that, I don't have any real preferences over what is used for forums, as long as it works and isn't a pain to use, I'd be happy. Guess it would depend on which would be easier to implement and maintain, and integrate with HabitRPG.

**Comment by horusofoz** [comment]
Discourse definitely looks cool. We'll doubtlessly review it in the future.
But for now, Drupal is the way.

**Comment by Pandoro** [comment]
@elssar I get all the fancy plus points :) And I don't want to doubt that it works. But what I am fearing a little, is that discourse will not he a habitrpg community forum, but rather will it be a fancy discourse show-case. We want the first, not the second :x Of course I get that it will not exactly be the way, but I just have the feeling that it will not be as neutral as other "more traditional" forum approaches. 

Looking at it for longer really makes me like the features, suggested topics, mentioning, search.... I am fearing we might make the wrong decision and cannot go back.... With Drupal we know what we get, with Discourse, not really... but we might get something way better. I have really got accustomed to GitHub, mentioning people with @, mentioning topics with #. This really rocks and you get that from Discourse... 

I would not like badges and points to be honest. I'm also sceptical about the trust system. So what I am wondering, can we strip these features down easily? Make it go so much to a basic level that we can just have the stuff we like and remove stuff we don't want? 

I hate myself for saying so, but I feel again like seriously considering this -.-; I think the eventual integration will be possible in discourse as well, albeit it might be easier in Drupal, with the open source community around habitrpg, someone will come around and just do it x)

**Comment by lefnire** [comment]
Maybe I should take a couple hours and hack on Discourse. I know what we can do with Drupal, so I'll see if we can do those on Discourse too. Incidentally, Habit being a Discourse showcase wouldn't be all that bad. Being a Derby showcase has actually gotten us some positive attention, and some developer interest

**Comment by Pandoro** [comment]
I think I fell in love with some of the Discourse features :p So I would
very much support that try ;)

**Comment by lefnire** [comment]
we decided on Drupal, and it's launched in a bare mim. form right now at http://community.habitrpg.com. I was playing with Discourse fo a bit, and it started to look like it would be a project in and of itself - so I switched to Drupal for convenience. After setting up the site, I realized it was the right decision because we can do a **ton** more with Drupal - FAQ, Blog, Podcast, Forum, static pages, etc. We already have many of those in disparate parts of the internet (FAQ b/w main site & Github, Blog on Tumblr, "Podcast" on Youtube, Forum on Reddit) - so being able to centralize all these things, not just Forum, makes Drupal a preferred solution to Discourse. We also have some Drupal talent in the community (props @wizonesolutions!) which will make this process go smoother.

I'm gonna close this ticket, let's follow progress on [Trello](https://github.com/HabitRPG/habitrpg-community) - and I'll be slowly-but-surely migrating various bits from around the internet into this community.habitrpg
